<a href="https://colab.research.google.com/github/2303A52060/2060/blob/main/sml_39_40.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Assignment-2:

Given the following dataset: X={1.3,1.5,2.6,2.6,3.2,3.9,4.2,3.7,3.10,3.0,11,6.7,1.9}. Compute the density of the data using a histogram with 5 bins.

Question 2:
dependent variable:
12,13,56,23,41,28,51
independent variable:
51,63,37,88,33,72,81

Find Mean, Median, Mode , Variance , Standard deviation , Range , Interquartile Range (IQR),  Skewness ,Kurtosis of x and y .

Question 3:

From the above data Find Mean, Median, Mode, Variance, Standard deviation, Range , Interquartile Range (IQR),  Skewness ,Kurtosis of each feature.


In [2]:
# Assignment 2
import numpy as np
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
# Example dataset
data = np.array([1.3, 1.5, 2.6, 2.6, 3.2, 3.9, 4.2, 3.7, 3.10, 3.0, 11, 6.7, 1.9])
# Mean
mean_value = np.mean(data)

# Median
median_value = np.median(data)

# Mode
mode_value, mode_count = stats.mode(data)

# Variance
variance_value = np.var(data, ddof=1)  # Sample variance (ddof=1)

# Standard Deviation
std_dev_value = np.std(data, ddof=1)  # Sample standard deviation (ddof=1)

# Range
range_value = np.ptp(data)

# Interquartile Range (IQR)
q75, q25 = np.percentile(data, [75, 25])
iqr_value = q75 - q25

# Skewness
skewness_value = stats.skew(data)

# Kurtosis
kurtosis_value = stats.kurtosis(data)

# Display the results
print(f"Mean: {mean_value:.2f}")
print(f"Median: {median_value:.2f}")
print(f"Variance: {variance_value:.2f}")
print(f"Standard Deviation: {std_dev_value:.2f}")
print(f"Range: {range_value:.2f}")
print(f"Interquartile Range (IQR): {iqr_value:.2f}")
print(f"Skewness: {skewness_value:.2f}")
print(f"Kurtosis: {kurtosis_value:.2f}")

Mean: 3.75
Median: 3.10
Variance: 6.66
Standard Deviation: 2.58
Range: 9.70
Interquartile Range (IQR): 1.30
Skewness: 1.87
Kurtosis: 2.95


solution:

Question 1: Compute the density of the data using a histogram with 5 bins
Given dataset:

𝑋
=
{
1.3
,
1.5
,
2.6
,
2.6
,
3.2
,
3.9
,
4.2
,
3.7
,
3.10
,
3.0
,
11
,
6.7
,
1.9
}
X={1.3,1.5,2.6,2.6,3.2,3.9,4.2,3.7,3.10,3.0,11,6.7,1.9}.

Determine the Range:

Minimum value:
1.3
1.3

Maximum value:
11
11

Range:
11
−
1.3
=
9.7
11−1.3=9.7

Determine Bin Width:

Number of bins: 5
Bin width:
9.7
5
=
1.94
≈
1.94
5
9.7
​
 =1.94≈1.94

Create Bins:

Bin 1:
[
1.3
,
3.24
)
[1.3,3.24)
Bin 2:
[
3.24
,
5.18
)
[3.24,5.18)
Bin 3:
[
5.18
,
7.12
)
[5.18,7.12)
Bin 4:
[
7.12
,
9.06
)
[7.12,9.06)
Bin 5:
[
9.06
,
11
)
[9.06,11)

Count the Data Points in Each Bin:

Bin 1:
{
1.3
,
1.5
,
2.6
,
2.6
,
1.9
,
3.0
}
{1.3,1.5,2.6,2.6,1.9,3.0} → 6 data points

Bin 2:
{
3.2
,
3.9
,
3.7
}
{3.2,3.9,3.7} → 3 data points

Bin 3:
{
4.2
}
{4.2} → 1 data point
Bin 4:

{
6.7
}
{6.7} → 1 data point

Bin 5:
{
11
}
{11} → 1 data point
Compute Density:

Total number of data points: 13

Bin densities (assuming equal bin width):

Density for Bin 1
=
6
13
×
1.94
≈
0.243
Density for Bin 1=
13×1.94
6
​
 ≈0.243

Density for Bin 2
=
3
13
×
1.94
≈
0.118
Density for Bin 2=
13×1.94
3
​
 ≈0.118

Density for Bin 3
=
1
13
×
1.94
≈
0.031
Density for Bin 3=
13×1.94
1
​
 ≈0.031

Density for Bin 4
=
1
13
×
1.94
≈
0.031
Density for Bin 4=
13×1.94
1
​
 ≈0.031

Density for Bin 5
=
1
13
×
1.94
≈
0.031
Density for Bin 5=
13×1.94
1
​
 ≈0.031
Question 2: Compute Statistical Measures for Dependent and Independent Variables

Dependent Variable:
𝑌
=
{
12
,
13
,
56
,
23
,
41
,
28
,
51
}
Y={12,13,56,23,41,28,51}

Independent Variable:
𝑋
=
{
51
,
63
,
37
,
88
,
33
,
72
,
81
}
X={51,63,37,88,33,72,81}

1. Dependent Variable Statistics:

Mean:
12
+
13
+
56
+
23
+
41
+
28
+
51
7
=
224
7
=
32
7
12+13+56+23+41+28+51
​
 =
7
224
​
 =32

Median: Sort data:
{
12
,
13
,
23
,
28
,
41
,
51
,
56
}
{12,13,23,28,41,51,56} → Median = 28

Mode: No repeating values, so no mode.

Variance:

Variance
=
∑
(
𝑥
𝑖
−

mean
)
2
𝑁
=
(
12
−
32
)
2
+
(
13
−
32
)
2
+
…
+
(
51
−
32
)
2
7

Variance=
N
∑(x
i
​
 −mean)
2

​
 =
7
(12−32)
2
 +(13−32)
2
 +…+(51−32)
2

​


=
400
+
361
+
576
+
16
+
81
+
169
+
361
7
=
1964
7
≈
280.57
=
7
400+361+576+16+81+169+361
​
 =
7
1964
​
 ≈280.57

Standard Deviation:
280.57
≈
16.75
280.57
​
 ≈16.75

Range:
56
−
12
=
44
56−12=44

Interquartile Range (IQR):
Q1 = 23
Q3 = 51
IQR = Q3 - Q1 = 51 - 23 = 28

Skewness: 0.1876
Kurtosis: -1.4768
2. Independent Variable Statistics:

Mean:
51
+
63
+
37
+
88
+
33
+
72
+
81
7
=
445
7
=
63.57
7
51+63+37+88+33+72+81
​
 =
7
445
​
 =63.57

Median: Sort data:
{
33
,
37
,
51
,
63
,
72
,
81
,
88
}
{33,37,51,63,72,81,88} → Median = 63

Mode: No repeating values, so no mode.

Variance:
Variance
=
∑
(
𝑥
𝑖
−

mean
)
2
𝑁
=
(
51
−
63.57
)
2
+
(
63
−
63.57
)
2
+
…
+
(
88
−
63.57
)
2
7

Variance=
N
∑(x
i
​
 −mean)
2

​
 =
7
(51−63.57)
2
 +(63−63.57)
2
 +…+(88−63.57)
2

​


=
161.47
+
0.32
+
160.61
+
603.61
+
927.37
+
129.87
+
613.67
7
≈
398.60
=
7
161.47+0.32+160.61+603.61+927.37+129.87+613.67
​
 ≈398.60

Standard Deviation:
398.60
≈
19.97
398.60
​
 ≈19.97

Range:
88
−
33
=
55
88−33=

Interquartile Range (IQR):
Q1 = 51
Q3 = 81
IQR = Q3 - Q1 = 81 - 51 = 30

Skewness: -0.1016
Kurtosis: -1.4265

Question 3: Compute the same statistics for each feature
Since the dataset provided in Question 2 already provides features (dependent and independent variables),